components/application-mgt/org.wso2.carbon.device.application.mgt.store.api/src/main/java/org/wso2/carbon/device/application/mgt/store/api/services/impl/CommentManagementAPIImpl.java

```diff
@@ -63,21 +61,19 @@ public Response getAllComments(
         List<Comment> comments = new ArrayList<>();
         try {
             PaginationRequest request = new PaginationRequest(offSet, limit);
-            if (request.validatePaginationRequest(offSet, limit)) {
+            if (uuid == null) {
+                return Response.status(Response.Status.NOT_FOUND).entity("Comments not found").build();
+            } else if (request.validatePaginationRequest(offSet, limit)) {
                 commentsManager.getAllComments(request, uuid);
                 return Response.status(Response.Status.OK).entity(comments).build();
             }
-        } catch (NotFoundException e) {
-            log.error("Not found exception occurs to uuid " + uuid + " .", e);
-            return Response.status(Response.Status.NOT_FOUND).entity("Application with UUID " + uuid + " not found")
-                    .build();
         } catch (CommentManagementException e) {
             String msg = "Error occurred while retrieving comments.";
             log.error(msg, e);
-            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
-                    .entity(" Internal server error occurs").build();
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(" Internal server error occurs")
```

Comment:
Suggestion. 
It is better to use error code with a simple resolution explanation. So I guess you can use the same error message that you are logging in the response as well.
"return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(msg).build()"

Reply:
Error messages were changed as suggested.